### DISCUSSION TITLE
Ability to check content collections without doing a build

### DISCUSSION THREAD

**Discussion by Princesseuh** [withastro/roadmap#759] [Proposal]
### Body

# Summary

It'd be neat to be able to run content collections checks as part of `astro check`, or in a separate command (maybe `astro sync` could do it, not sure)

# Background & Motivation

In CI, it can be useful to check before doing a full build, assuming the checking is faster than the full build (which it should be in this case, I'd assume)

# Goals

- Be able to check all files quickly without doing the entire build process. This could includes more extensive error reporting possibly as well

**Comment by delucis** [comment]
I believe current checking also only reports the first error? So this could be an opportunity for a more bird’s eye view of all the errors when you have a lot of them.

**Comment by marekbrze** [comment]
It would be super useful. Especially when using `astro check`. I understand that right now we won't get any error without doing build or navigating to the specific path that queries that faulty file.
